nasa-gibs/worldview
# Support comparison mode in mobile

## Description

Fixes #3188  .

- [x] Allow "official support" to comparison mode in mobile. Comparison mode will be limited to `swipe` in mobile and will use the same sidebar A/B tabs along with a new A/B toggle button.
- [x] Add initial mobile screen size check to config on load as `config.initialIsMobile` to help determine location state changes for compare permalinks opened in mobile
- [x] Add e2e mobile-compare tests and minor update on a few others.
- [x] Minor cleanup removing unnecessary dom depth, remove unused function params, and random spell fixes from adding a new extension to VS Code `Code Spell Checker`

## How To Test

1. Activate compare mode in mobile, verify activate/deactivate, verify desktop permalinks with compare mode open in mobile including a desktop `spy` compare permalink should open as a `swipe` when opened on a mobile device


## PR Submission Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/master/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview
